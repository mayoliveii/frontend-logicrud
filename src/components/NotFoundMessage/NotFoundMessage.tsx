import { Box, Flex, Heading, Text } from '@chakra-ui/react';
import { CloseIcon } from '@chakra-ui/icons';

interface ErrorProps {
  title: string;
  description: string;
}

export default function NotFoundMessage({ title, description }: ErrorProps) {
  return (
    <Box textAlign="center" py={10} px={6} m={20}>
      <Box display="inline-block">
        <Flex
          flexDirection="column"
          justifyContent="center"
          alignItems="center"
          bg={'red.500'}
          rounded={'50px'}
          w={'55px'}
          h={'55px'}
          textAlign="center">
          <CloseIcon boxSize={'20px'} color={'white'} />
        </Flex>
      </Box>
      <Heading as="h2" size="xl" mt={6} mb={2}>
        {title}
      </Heading>
      <Text color={'gray.500'}>
        {description}
      </Text>
    </Box>
  )
}