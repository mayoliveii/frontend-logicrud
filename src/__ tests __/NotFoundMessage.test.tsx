import { render } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import NotFoundMessage from '../components/NotFoundMessage/NotFoundMessage';
import '@testing-library/jest-dom';

describe('NotFoundMessage', () => {
  it('renders Error component with given title and description', () => {
    const { getByText, asFragment } = render(
      <ChakraProvider>
        <NotFoundMessage title="Test Error" description="This is a test error" />
      </ChakraProvider>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText('Test Error')).toBeInTheDocument();
    expect(getByText('This is a test error')).toBeInTheDocument();
  });
});