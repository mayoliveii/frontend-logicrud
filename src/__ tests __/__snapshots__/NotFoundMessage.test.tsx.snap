// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundMessage renders Error component with given title and description 1`] = `
<DocumentFragment>
  <div
    class="css-9gtaw2"
  >
    <div
      class="css-1baulvz"
    >
      <div
        class="css-f5kz2b"
      >
        <svg
          class="chakra-icon css-1x7z2yw"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <h2
      class="chakra-heading css-1wib1wc"
    >
      Test Error
    </h2>
    <p
      class="chakra-text css-q9k0mw"
    >
      This is a test error
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
